indigo/docs/guides/howto-hello-indigo-tyrian.md

```diff
@@ -0,0 +1,418 @@
+website surrounding your game with no interaction, but it would be much more
+useful to get them both talking to each other... this is what we're doing next!
+
+## Tyrian to Indigo Communication
+
+The first thing we'll do is get Tyrian to request that Indigo add a new dot. To
+do this we'll need to add an `OnClick` event to our button, which will tell
+Indigo what to do.
+
+In `HelloTyrian.scala`, we'll need a new message type. Add a new message type called AddDot like so:
+
+```diff
+   enum Msg:
+     case StartIndigo extends Msg
++    case AddDot      extends Msg
+```.
+
+Next we'll need to deal with that message in Tyrian, by adding a case to `update`
+like this:
+
+```diff
+     msg match
++      case Msg.AddDot =>
++        (model, model.bridge.publish(IndigoGameId(gameDivId), 0))
+       case Msg.StartIndigo =>
+```
+
+The final part on the Tyrian side is to hook the button up to fire the `AddDot`
+message. To do this add an `OnClick` attribute to our button like so:
+
+```diff
+       div(`class` := "btn")(
+-        button()("Click me")
++        button(onClick(Msg.AddDot))("Click me")
+       )
+```
+
+This will now send an event to Indigo with an integer message, which in this case
+we've set to zero. Now we just need to modify Indigo to receive and
+process that message. To do this change the `MouseEvent.Click` line in
+`HelloIndigo.scala` with the following:
+
+```diff
+   ): GlobalEvent => Outcome[Model] = {
+-    case MouseEvent.Click(pt) =>
++    case tyrianSubSystem.TyrianEvent.Receive(msg) =>
++      val pt               = Point(100, 100)
+       val adjustedPosition = pt - model.center
+```
+
+We've also added a line below that to set a `pt` variable as we now longer have
+a mouse position. For now, we've set that to 100,100 like so which is just a
+fixed point for the dots to rotate around.
+
+Once more build the project in mill or sbt and then run `yarn start`. This time
+you'll notice that clicking the game doesn't do anything, but clicking the
+button adds a new rotating dot.
+
+We've now successfully gotten Tyrian to talk to Indigo. In the next part we'll
+be doing the opposite, and adding a counter for the number of dots.
+
+## Indigo to Tyrian Communication
+
+In this part we're going to store a count of the number of dots that Indigo is
+currently displaying. To do this, first update the model in `HelloTyrian.scala`
+so that it has a `count` property which is initialised to zero, like so:
+
+```diff
+-final case class TyrianModel(bridge: TyrianIndigoBridge[Int])
++final case class TyrianModel(bridge: TyrianIndigoBridge[Int], count: Int)
+object TyrianModel:
+-  val init: TyrianModel
++  val init: TyrianModel =
++    TyrianModel(TyrianIndigoBridge(), 0)
+```
+
+We'll need to display that count on the website. To do this, we simply modify our
+counting `div` to display what's in the `model` like so
+
+```diff
+-      div(`class` := "counter")(),
++      div(``class`` := "counter")(model.count.toString)
```

Comment:
Do you have double back ticks? Might be better to use one of the aliases? Valid options:

- ``class``
- `cls`
- `className`
- `_class`

Reply:
The double backticks are to escape it for markdown - this will render as
```scala
``class``
```

Reply:
(or at least it does in my Markdown previewer)